substreams/src/parser.rs

```diff
@@ -0,0 +1,53 @@
+use pest::{iterators::Pair, Parser};
+use pest_derive::Parser;
+
+#[derive(Parser)]
+#[grammar = "parser_rule.pest"]
+pub struct EParser;
+
+pub fn evaluate_expression(keys: Vec<String>, input: &str) -> bool {
```

Comment:
No need to receive a owned copy of a `Vec<String>`. We should accept variant of `String` like `&str` too. I would change to
```suggestion
pub fn evaluate_expression<K: AsRef<str>, I: AsRef<str>>(keys: &[K], input: I) -> bool {
```

And this can be used now with different variations of `String`, `&String` and `&str`:

```
    evaluate_expression(&vec!["a", "b", "c"], "d");
    evaluate_expression(
        &vec!["a".to_string(), "a".to_string(), "a".to_string()],
        "d",
    );

    let keys = vec!["a".to_string(), "a".to_string(), "a".to_string()];
    let keys_ref = keys.iter().map(|x| x).collect::<Vec<&String>>();
    evaluate_expression(&keys_ref, "d");
 ```

Reply:
Let's also find a more meaningful name for this function, it will be the only one exposed IMO so we better find it a very good clear name.